File: src/main/java/entity/BossDamage.java
package com.wenkrang.nian_beast.entity;

import com.wenkrang.nian_beast.NianBeast;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.scheduler.BukkitRunnable;

import static com.wenkrang.nian_beast.entity.PlayerJoin.AutoSetTarget;
import static com.wenkrang.nian_beast.NianBeast.isShutdown;

public class BossDamage implements Listener {
    @EventHandler
    public static void OnBossDamage (PlayerJoinEvent event) {
        new BukkitRunnable() {
            /**
             * 运行方法
             */
            @Override
            public void run() {
                // 如果已经关闭，则取消任务
                if (isShutdown()) {
                    cancel();
                }
                // 如果玩家在线
                if (event.getPlayer().isOnline()) {
                    // 创建一个BukkitRunnable任务
                    new BukkitRunnable() {
                        @Override
                        public void run() {
                            AutoSetTarget(40, "nian_beastthree", event.getPlayer());
                        }
                    }.runTaskLater(NianBeast.getPlugin(NianBeast.class), 0);

                } else {
                    // 如果玩家不在线，则取消任务
                    cancel();
                }
            }
        }.runTaskTimerAsynchronously(NianBeast.getPlugin(NianBeast.class), 0, 40);

    }
}

File: src/main/scala/entity/BossDamage.scala
package com.wenkrang.nian_beast.entity

import com.wenkrang.nian_beast.NianBeast
import com.wenkrang.nian_beast.NianBeast.isShutdown
import com.wenkrang.nian_beast.entity.PlayerJoin.AutoSetTarget
import org.bukkit.event.{EventHandler, Listener}
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.plugin.java.JavaPlugin
import org.bukkit.scheduler.BukkitRunnable

class BossDamage extends Listener {
  @EventHandler def OnBossDamage(event: PlayerJoinEvent): Unit = {
    new BukkitRunnable() {
      /**
       * 运行方法
       */
      override def run(): Unit = {
        // 如果已经关闭，则取消任务
        if (isShutdown) cancel()
        // 如果玩家在线
        if (event.getPlayer.isOnline) {
          // 创建一个BukkitRunnable任务
          new BukkitRunnable() {
            override def run(): Unit = {
              AutoSetTarget(40, "nian_beastthree", event.getPlayer)
            }
          }.runTaskLater(JavaPlugin.getPlugin(classOf[NianBeast]), 0)
        }
        else {
          // 如果玩家不在线，则取消任务
          cancel()
        }
      }
    }.runTaskTimerAsynchronously(JavaPlugin.getPlugin(classOf[NianBeast]), 0, 40)
  }
}